centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java

```diff
@@ -829,10 +829,16 @@ private void handleAsyncReply(Protocol.Reply reply) {
             String channel = push.getChannel();
             if (push.getType() == Protocol.Push.PushType.PUBLICATION) {
                 Protocol.Publication pub = Protocol.Publication.parseFrom(push.getData());
+                ClientInfo info = new ClientInfo();
+                info.setClient(pub.getInfo().getClient());
+                info.setUser(pub.getInfo().getUser());
+                info.setConnInfo(pub.getInfo().getConnInfo().toByteArray());
+                info.setChanInfo(pub.getInfo().getChanInfo().toByteArray());
                 Subscription sub = this.getSub(channel);
                 if (sub != null) {
                     PublishEvent event = new PublishEvent();
                     event.setData(pub.getData().toByteArray());
+                    event.setInfo(info);
```

Comment:
Should not be set if it's not set in incoming Publication - i.e. should be null, not an empty ClientInfo object as done here.